octobot_backtesting/independent_backtesting.py

```diff
@@ -0,0 +1,336 @@
+            BOT_REPORT = "bot_report"
+            CHART_IDENTIFIERS = "chart_identifiers"
+            report = {
+                SYMBOL_REPORT: [],
+                BOT_REPORT: {},
+                CHART_IDENTIFIERS: []
+            }
+
+            refenrence_market = get_reference_market(self.backtesting_config)
+            try:
+                trading_mode = get_activated_trading_mode(self.backtesting_config).get_name()
+            except ConfigTradingError as e:
+                self.logger.error(e)
+                trading_mode = "Error when reading trading mode"
+            for exchange_id in self.exchange_manager_ids:
+                exchange_manager = get_exchange_manager_from_exchange_id(exchange_id)
+                _, profitability, _, market_average_profitability, _ = get_profitability_stats(exchange_manager)
+                min_timeframe = find_min_time_frame(get_watched_timeframes(exchange_manager))
+                exchange_name = get_exchange_name(exchange_manager)
+                for symbol in self.symbols_to_create_exchange_classes[exchange_name]:
+                    market_delta = self._get_market_delta(symbol, exchange_manager, min_timeframe)
+                    report[SYMBOL_REPORT].append({symbol: market_delta * 100})
+                    report[CHART_IDENTIFIERS].append({
+                        "symbol": symbol,
+                        "exchange_id": exchange_id,
+                        "exchange_name": exchange_name,
+                        "time_frame": min_timeframe.value
+                    })
+
+                report[BOT_REPORT] = {
+                    "profitability": profitability,
+                    "market_average_profitability": market_average_profitability,
+                    "reference_market": refenrence_market,
+                    "end_portfolio": get_portfolio(exchange_manager),
+                    "starting_portfolio": get_origin_portfolio(exchange_manager),
+                    "trading_mode": trading_mode
+                }
+                # TODO: handle multi exchange reports
+                return report
+        except ImportError as e:
+            self._log_import_error()
+            raise e
+
+    def _adapt_config(self):
+        from octobot_trading.constants import CONFIG_TRADER_RISK, CONFIG_TRADING, CONFIG_SIMULATOR, \
+            CONFIG_STARTING_PORTFOLIO, CONFIG_SIMULATOR_FEES,CONFIG_TRADER_REFERENCE_MARKET
+        self.backtesting_config[CONFIG_TRADING][CONFIG_TRADER_RISK] = self.risk
+        self.backtesting_config[CONFIG_TRADING][CONFIG_TRADER_REFERENCE_MARKET] = self._find_reference_market()
+        self.backtesting_config[CONFIG_SIMULATOR][CONFIG_STARTING_PORTFOLIO] = self.starting_portfolio
+        self.backtesting_config[CONFIG_SIMULATOR][CONFIG_SIMULATOR_FEES] = self.fees_config
+        self._add_config_default_backtesting_values()
+        self._add_crypto_currencies_config()
+
+    def _find_reference_market(self):
```

Comment:
Is this a backtesting specific code ?

Reply:
Yes, in other cases we should use the config ref market